docs/contributor/performance-tests/metrics/metrics.md

```diff
@@ -0,0 +1,690 @@
+    prometheus.io/scrape: "true"
+    prometheus.io/scheme: "http"
+  labels:
+    app.kubernetes.io/name: telemetry-metric-agent
+  name: telemetry-metric-agent-metrics
+  namespace: kyma-system
+spec:
+  clusterIP: None
+  ports:
+  - name: http-metrics
+    port: 8888
+    protocol: TCP
+    targetPort: 8888
+  selector:
+    app.kubernetes.io/name: telemetry-metric-agent
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+## Testcases
+
+### Assumptions
+
+We tweak metrics and series value with 10 labels in each metric data point. The test is executed for 1 hour to have a stabilized output. Additionally, the load is scaled slowly to prevent spikes which might cause OOMs of metric agent.
+
+We identified following test cases:
+1. Multiple pods all running on a single node and export metrics (to find how many workloads supported)
+2. Workload generating huge amount of metrics (To understand how scraping works when the workload exposes several MB of metrics)
+3. Have multiple workloads across different nodes (To understand prometheus SDS behaviour with multiple services)
+4. Verify sample limit test
+5. Multiple pods and services scraped from multiple receivers
+
+### Multiple pods all running on a single node and export metrics
+Setup:
+
+Avalanche load generator configuration per pod:
+- Metric count: 500
+- Metric series: 20
+- Number of labels: 10
+
+To simulate a smooth ramp-up and avoid huge data flood at the beginning, load generator was started with 5 instances, instance count was increased at 5 instances every minute until peak was reached.
+
+This test was executed on a single node installation to determine the limits of a single metric agent, after the load generator reached `28` instances, metric agent hits the limits and stays stable with this setup. Any load above this setup would cause memory limiter to refuse incoming traffic.
+
+The graph below shows the average metric points processed by the `prometheus metric receiver` per second, agent reaches an average of `14K metric points/sec`
+![Peak accepted metric points](./assets/overall-peak-metric-point.jpg)
+
+Following graph shows the refused metric points from the configured `memory_limiter`, with traffic above 14K metric points/sec results in  metrics being refused by the `memory_limiter`. 
+![Peak refused metric points](./assets/overall-peak-metric-point-refused.jpg)
+
+Following graph shows memory utilization of metric agent during test phase.
+![Peak memory utilization of metric agent](./assets/overall-peak-memory.jpg)
+
+
+Following graph shows CPU utilization of metric agent during test phase.
+![Peak CPU utilization of metric agent](./assets/overall-peak-cpu.jpg)
+
+
+### Workload generating huge amount of metrics
+
+This scenario tests a single metric endpoint with huge amount of data, Avalanche load generator deployed with a single pod instance and configured with 1000 distinct metrics and 20 metric series for each metric with 10 labels.
+
+Setup:
```

Comment:
line 306 and 309ff seem to convey the exact same information. Why do we present the same content twice?

Reply:
Reworked